src/tiles/text-snippet-block.ts

```diff
@@ -0,0 +1,116 @@
+import { css, CSSResultGroup, html, LitElement, nothing } from 'lit';
+import { customElement, property } from 'lit/decorators.js';
+import { unsafeHTML } from 'lit/directives/unsafe-html.js';
+import { map } from 'lit/directives/map.js';
+import { join } from 'lit/directives/join.js';
+import DOMPurify from 'dompurify';
+
+@customElement('text-snippet-block')
+export class TextSnippetBlock extends LitElement {
+  // The default snippets value here is just for testing's sake; will be removed.
+  @property({ type: Array }) snippets?: string[] = [
+    'this is a text {{{snippet}}} block with potentially',
+    'multiple {{{snippets}}} and such',
+    'but the {{{snippet}}} block may be quite long perhaps',
+    'depending on how many {{{snippet}}} matches there are',
+    'there may be multiple lines of {{{snippets}}} to show',
+    'but each {{{snippet}}} should be relatively short',
+  ];
+
+  @property({ type: String }) viewSize: string = 'desktop';
+
+  render() {
+    const viewSizeClass = this.viewSize === 'grid' ? 'grid' : 'list';
+
+    return html`
+      <div id="container" class=${viewSizeClass}>
+        &hellip; ${this.ellipsisJoinedSnippets ?? nothing} &hellip;
+      </div>
+
+      ${this.viewSize === 'grid' ? html`<div id="separator"></div>` : nothing}
+    `;
+  }
+
+  /**
+   * An array of HTML templates derived from the snippets, with ellipses inserted
+   * between each pair of snippets.
+   */
+  private get ellipsisJoinedSnippets() {
+    const sanitizeOptions = { ALLOWED_TAGS: ['mark'] };
+
+    return join(
+      map(
+        this.snippets,
+        s => html`
+          <span>
+            ${unsafeHTML(
+              DOMPurify.sanitize(
+                TextSnippetBlock.markMatches(s),
```

Comment:
Q: any particular reason why we need/want this fn to be static?

Reply:
My thinking was that it's an instance-independent util fn and probably not useful outside of this class (though I may be wrong about that!).

Reply:
Refactored this into a getter that produces marked versions of a component's snippets.